4.classify-single-cell-phenotypes/classification_utils.py

```diff
@@ -25,28 +25,30 @@ def get_probas_dataframe(
         model to use for plate feature classification
     feature_type : str
         type of features to use for classification.
-        CP, DP, or CP_and_DP
+        CP, DP, CP_and_DP, CP_areashape_only, CP_zernike_only
 
     Returns
     -------
     pd.DataFrame
         dataframe with single-cell probabilities for classes from given model
     """
+    
+    all_cols = plate_features.columns.to_list()
 
     # determine which feature columns should be loaded depending on feature type
-    # if there is no "and" we can use feature type as prefix
-    if feature_type != "CP_and_DP":
-        cols_to_load = [
-            col
-            for col in plate_features.columns.to_list()
-            if f"{feature_type}__" in col
-        ]
-    # if there is an "and" we should use all features (which all have "P__" prefix)
-    else:
-        cols_to_load = [col for col in plate_features.columns.to_list() if "P__" in col]
+    if "CP" in feature_type:
+        feature_cols = [col for col in all_cols if "CP__" in col]
+        if "zernike_only" in feature_type:
+            feature_cols = [col for col in feature_cols if "Zernike" in col]
+        if "areashape_only" in feature_type:
+            feature_cols = [col for col in feature_cols if "AreaShape" in col]
+        if "_and_DP" in feature_type:
+            feature_cols = [col for col in all_cols if "P__" in col]
```

Comment:
```suggestion
        feature_cols = [col for col in all_cols if "CP__" in col]
        if "zernike_only" in feature_type:
            feature_cols = [col for col in feature_cols if "Zernike" in col]
        elif "areashape_only" in feature_type:
            feature_cols = [col for col in feature_cols if "AreaShape" in col]
        elif "_and_DP" in feature_type:
            feature_cols = [col for col in all_cols if "P__" in col]
        else:
            raise ValueError("Invalid feature type")
```
This question focuses more on the design of this block. Applying multiple `if` statements is exhaustive. Meaning, if the `feature_type` were to be found within the multiple `if` statements (e.g `zernike_only`) the subsequent `if` statements will still be executed. For example if there is a string `areashape_only_and_DP`, the `feature_cols` variable under the `areas_shape_only` will be overwritten by the `feature_cols` under the `_and_DP` `if` statement. A good way to avoid this is using `elif`. Where it will stop once it finds the condition. 

I do not know if this is the expected behavior! feel free to reject!

Reply:
I think using the `elif` here makes sense, thanks for the suggestion! I will incorporate on my end. We definitely do not want to raise an error if none are found because that is not expected behavior.